signac/contrib/project.py

```diff
@@ -95,6 +95,20 @@ def find_job_ids(self, filter=None, doc_filter=None):
         return self._collection._find(filter)
 
 
+class ProjectConfig(Config):
+    """Extends the project config to make it immutable."""
+    def __init__(self, *args, **kwargs):
+        self._mutable = True
+        super(ProjectConfig, self).__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        if not self._mutable:
+            warnings.warn("Modifying the project configuration is deprecated "
```

Comment:
1. Let's use the standard language for other deprecations, i.e., "deprecated as of version 1.3 ..."
2. I'd try to make clear that modifying the project configuration *in this way* is deprecation.

So maybe something like this:

> Modifying the project configuration after project initialization is deprecated as of version 1.3 and will be removed in version 2.0.